### PR TITLE
Essentials.Email refactored to interface and singleton

### PR DESCRIPTION
**Pull request by moljac**
### Description of Change ###

Implements #

```csharp
	public interface IEmail
	{
		bool IsComposeSupported { get; }

		Task ComposeAsync();

		Task ComposeAsync(string subject, string body, params string[] to);

		Task ComposeAsync(EmailMessage message);
	}
```

### Additions made ###

* Adds 

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the WinUI, Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

**Comment by rmarinho** [comment]
You have to look the error before 

```
  D:\a\_work\1\s\src\Essentials\src\Email\Email.uwp.cs(14,45): error CS0736: 'EmailImplementation' does not implement instance interface member 'IEmail.IsComposeSupported'. 'EmailImplementation.IsComposeSupported' cannot implement the interface member because it is static. [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]
```

**Comment by moljac** [comment]
> You have to look the error before
> 
> ```
>   D:\a\_work\1\s\src\Essentials\src\Email\Email.uwp.cs(14,45): error CS0736: 'EmailImplementation' does not implement instance interface member 'IEmail.IsComposeSupported'. 'EmailImplementation.IsComposeSupported' cannot implement the interface member because it is static. [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]
> ```

Windows again. I am on Mac, so this is skipped.

**Review by rmarinho — CHANGES_REQUESTED** [review]
build failing

```
(CoreCompile target) -> 
  D:\a\_work\1\s\src\Essentials\src\Email\Email.uwp.cs(63,18): error CS1061: 'string[]' does not contain a definition for 'List' and no accessible extension method 'List' accepting a first argument of type 'string[]' could be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]

    749 Warning(s)
    1 Error(s)
```

**Comment by moljac** [comment]
> build failing
> 
> ```
> (CoreCompile target) -> 
>   D:\a\_work\1\s\src\Essentials\src\Email\Email.uwp.cs(63,18): error CS1061: 'string[]' does not contain a definition for 'List' and no accessible extension method 'List' accepting a first argument of type 'string[]' could be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Essentials\src\Essentials.csproj]
> 
>     749 Warning(s)
>     1 Error(s)
> ```

Fixed:

https://github.com/dotnet/maui/pull/4623/commits/1d55f4a0ad0f8241992292ee296b83ea990c4f87

**Review by Redth — APPROVED** [review]

File: src/Essentials/src/Email/Email.shared.cs [hunk 1]
```diff
@@ -2,9 +2,23 @@
...
+using Microsoft.Maui.Essentials;
+using Microsoft.Maui.Essentials.Implementations;
 
 namespace Microsoft.Maui.Essentials
 {
+	public interface IEmail
+	{
+		bool IsComposeSupported { get; }
+		
+		Task ComposeAsync();
```
Comment (rmarinho) [inline src/Essentials/src/Email/Email.shared.cs:15]: What does this task do @Redth ?

Comment (Redth) [reply src/Essentials/src/Email/Email.shared.cs:15]: Blank new email, with no to/subject/etc.

**Review by Redth — APPROVED** [review]

**Pull request merged**